hotkeys: Disable "w" hotkey when settings dropdown is open.

2208Abhinav opened the pull request:
Settings dropdown covers users searchbar when it's open. In that
case we don't want to initialize user search when "w" is pressed
with dropdown menu open on top of it.

Fixes #11990.

@timabbott please review this pr :)

YJDave commented:
@2208Abhinav Thanks for working on this!
`./tools/test-js-with-node --coverage` tests are failing in PR. You should checkout tests written in `frontend_tests/node_tests/hotkey.js` and update tests there according to changes you made to the function.

2208Abhinav commented:
@YJDave thank you for pointing this out :).
I went out for a medical checkup and then forgot to add tests. Now the tests are passing.

timabbott commented:
This is way too tactical.  What we should be doing is:
* For all `message_view_only` hotkeys add a conditional, just after this block: `    if (menu_dropdown_hotkeys.indexOf(event_name) !== -1) {                                                                                                               ` that does a `return false` if the gear menu is open.
* Add a proper node test covering this code case using the mocking strategy used in `hotkey.js`.

2208Abhinav commented:
@timabbott as you mentioned **{for all `message_view_only` do `return false` if `gear_menu is open`}**, I changed the code but it didn't work as expected. But I changed the code to  **{for all `not message_view_only` do `return false` if `gear_menu is open`}** and it works fine. I have updated the pr for you to review and test.
And as far as I understand the second case is correct because we want to disable `q`, `w` and all other hotkeys that are out message view but we want want to keep `r` (and other in message view hotkeys) to be enabled when gear menu is open.

timabbott commented:
Hmm, I'm not sure this is the right fix.  @lonerz @showell do you have an understanding of why `q` and `w` are not `message_view_only` hotkeys?  Certainly they are not relevant hotkeys with an overlay open...

lonerz commented:
They should be `message_view_only`. I don't remember why we kept them otherwise, but on review currently, they are only used to activate search bars, which is only relevant in the message view.

2208Abhinav commented:
@timabbott so should I do a pr changing them to `message_view_only`. But is guess this type of change will lead to other side effects right?

timabbott commented:
@2208Abhinav what I would do is update this PR to be a series of commit starting with one changing those to `message_view_only`, and continuing with the gear menu change I suggested above.

timabbott commented:
Closing in favor of #13809.  Thanks for your work on this @2208Abhinav!
